internal/dot/graph.go

```diff
@@ -177,27 +253,26 @@ func (g *Group) String() string {
 	return fmt.Sprintf("[type=%v group=%v]", g.Type.String(), g.Name)
 }
 
-// Attributes composes and returns a string to style the Param's sublabel.
-func (p *Param) Attributes() string {
-	if p.Name != "" {
-		return fmt.Sprintf(`<BR /><FONT POINT-SIZE="10">Name: %v</FONT>`, p.Name)
-	}
-	return ""
-}
-
-// Attributes composes and returns a string to style the Result's sublabel.
+// Attributes composes and returns a string of the Result node's attributes.
 func (r *Result) Attributes() string {
 	switch {
 	case r.Name != "":
-		return fmt.Sprintf(`<BR /><FONT POINT-SIZE="10">Name: %v</FONT>`, r.Name)
+		return fmt.Sprintf(`label=<%v<BR /><FONT POINT-SIZE="10">Name: %v</FONT>>`, r.Type, r.Name)
 	case r.Group != "":
-		return fmt.Sprintf(`<BR /><FONT POINT-SIZE="10">Group: %v</FONT>`, r.Group)
+		return fmt.Sprintf(`label=<%v<BR /><FONT POINT-SIZE="10">Group: %v</FONT>>`, r.Type, r.Group)
 	default:
-		return ""
+		return fmt.Sprintf(`label=<%v>`, r.Type)
 	}
 }
 
-// Attributes composes and returns a string to style the Group's sublabel.
+// Attributes composes and returns a string of the Group node's attributes.
 func (g *Group) Attributes() string {
-	return fmt.Sprintf(`<BR /><FONT POINT-SIZE="10">Group: %v</FONT>`, g.Name)
+	attr := fmt.Sprintf(`shape=diamond label=<%v<BR /><FONT POINT-SIZE="10">Group: %v</FONT>>`, g.Type, g.Name)
+	switch g.State {
+	case pointOfFailure:
+		attr += " color=red"
```

Comment:
this appears to be repeating some of the logic from `if state == 1`. The `Color` method would help here too.